src/operator/tensor/indexing_op.cc

```diff
@@ -95,6 +95,77 @@ Examples::
 .add_argument("weight", "NDArray-or-Symbol", "The embedding weight matrix.")
 .add_arguments(EmbeddingParam::__FIELDS__());
 
+NNVM_REGISTER_OP(_contrib_SparseEmbedding)
```

Comment:
Stupid question: Why are we registering a new operator here instead of extending the existing embedding op ?

Reply:
Good question! Because if we override the existing op, we don't know whether to infer rsp grad or dense grad for weight, because the weight ndarray is not visible for backward pass.